lib/core/macros.nim

```diff
@@ -1418,6 +1418,12 @@ proc boolVal*(n: NimNode): bool {.compileTime, noSideEffect.} =
   if n.kind == nnkIntLit: n.intVal != 0
   else: n == bindSym"true" # hacky solution for now
 
+when defined(nimMacrosGetNodeId):
+  proc nodeID*(n: NimNode): int {.magic: NodeId.}
+    ## Returns the id of ``n``, when the compiler has been compiled
+    ## with the flag ``-d:useNodeids``, otherwise returns ``-1``. This
+    ## proc is for for the purpose to debug the compiler only.
```

Comment:
Typo: "for for".

Also: why is it in macros.nim then? This could be an import of its own.

Reply:
I put it in ``macros.nim`` because I never thought about putting it anywhere else. Also why should it be in it's own module? It is not like that it is causing a lot of overhead for the compilation.